src/components/Button/Button.styled.js

```diff
@@ -0,0 +1,29 @@
+import styled,{css} from 'styled-components';
+
+const DefaultStyledButton = styled.button`
+    background-color: black;
+    color: white;
+    font-size: 1rem;
+    padding: 0.375rem 0.75rem;
+    border-radius: 0.3rem;
+    margin:2px 2px;
+    cursor: pointer;
+    &:hover {
+        opacity: 0.7;
+    }
+    &:disabled {
+        opacity: .7;
+        cursor: default;
+        pointer-events: none;
+    }
+`
+const StyledButton = styled(DefaultStyledButton)`
+    ${(props)=>props.theme.buttons[props.variant]};
+    ${(props)=>props.theme.buttonsSizes[props.size]};
+    background:${(props)=> props.active ? props.theme.buttonsActive[props.variant] : null};
+    opacity:${(props)=>props.disabled ? .55 : null}; // to opacity mi nie działa jak mam wyzej odkomentowane style, opacity nie powinno sienadpisac?
```

Comment:
Nie wiem, o który dokładnie chodzi, ale zobacz, że tam masz dodatkowe opcje tj. `:disabled` lub `:hover`, które zwiększają specyficzność i one są wtedy ważniejsze (to one są brane pod uwagę).
![image](https://user-images.githubusercontent.com/55095257/164385404-9849a3aa-20b2-441e-9c2d-9211a8c4220c.png)